src/mrinufft/operators/base.py

```diff
@@ -275,6 +278,55 @@ def __init_subclass__(cls):
         if backend := getattr(cls, "backend", None):
             cls.interfaces[backend] = (available, cls)
 
+    def check_shape_op(self, image, ksp=None):
+        """Validate that the shape of the provided image matches the expected shape.
+
+        This validation is defined by the operator during initialization.
+
+        Parameters
+        ----------
+        image : np.ndarray or Tensor
+
+        Returns
+        -------
+        None
+            This function does not return any value. It raises a ValueError if the
+            image shape does not match the expected shape.
+
+        """
+        image_shape = image.shape[-len(self.shape) :]
+
+        if image_shape != self.shape:
+            raise ValueError(
+                f"Image shape {image_shape} is not compatible "
+                f"with the operator shape {self.shape}"
+            )
+
+    def check_shape_adj_op(self, image, ksp=None):
+        """Validate that the shape of the provided image matches the expected shape.
+
+        This validation is defined by the operator during initialization.
+
+        Parameters
+        ----------
+        image : np.ndarray or Tensor
+
+        Returns
+        -------
+        None
+            This function does not return any value. It raises a ValueError if the
+            image shape does not match the expected shape.
+        """
+        image_samples = (
+            image.n_samples if hasattr(image, "n_samples") else image.shape[-1]
+        )
```

Comment:
I dont know if it is image. Why are we comparing image.n_samples here? shouldnt we just check image.shape?

Reply:
I think you meant kspace?